sql/core/src/test/scala/org/apache/spark/sql/SparkSessionBuilderSuite.scala

```diff
@@ -589,4 +589,13 @@ class SparkSessionBuilderSuite extends SparkFunSuite with Eventually {
       assert(session.conf.get(e._1) == e._2.toString)
     }
   }
+
+  test("SPARK-50222: Support spark.submit.appName") {
+    val session = SparkSession.builder()
```

Comment:
Does this test case cover the scenarios exemplified in the PR description? 

I copied this test case to the master branch and changed 

```
.config("spark.submit.appName", "newAppName")
```
to  

```
.config("spark.app.name", "newAppName")
```

and I found that the test still passed.

Reply:
BTW, I admit that the current test case is too simplified to verify the new logic only, @LuciferYang , of course. Thank you for review. 😄

Reply:
Got it ~